### PR TITLE
[ticket/16143] Add core events for move topics

### PR DESCRIPTION
**Pull request by OxCom**
Checklist:

- [ ] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [ ] Tests pass
- [ ] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [ ] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket

https://tracker.phpbb.com/browse/PHPBB3-16143

### CODE REVIEW

**Comment by dark-1** [comment]
@OxCom ,

Force push the last commit with `PHPBB3-16143` at it's end,
just like it's previous on...

Best regards 👍

**Comment by dark-1** [comment]
!set Event

**Comment by 3D-I** [comment]
> 
> 
> !set Event

You can do it only in your PRs.

**Comment by dark-1** [comment]
> > !set Event
> 
> You can do it only in your PRs.

Yea! Got it...
😉

File: phpBB/includes/functions_admin.php [hunk 1]
```diff
@@ -543,6 +543,20 @@ function move_topics($topic_ids, $forum_id, $auto_sync = true)
 		$topic_ids = array($topic_ids);
 	}
 
+	/**
+	 * Perform additional actions before topics move
+	 *
+	 * @event core.move_topics_before
+	 * @var	array	topic_ids	Array of the moved topic ids
+	 * @var	string	forum_id	The forum id from where the topics are moved
+	 * @since 3.2.x
```
Comment (dark-1) [inline phpBB/includes/functions_admin.php]: Use :
```php
@since 3.2.9-RC1
```

File: phpBB/includes/functions_admin.php [hunk 2]
```diff
@@ -593,6 +607,22 @@ function move_topics($topic_ids, $forum_id, $auto_sync = true)
...
 	unset($table_ary);
 
+	/**
+	 * Perform additional actions after topics move
+	 *
+	 * @event core.move_topics_after
+	 * @var	array	topic_ids	Array of the moved topic ids
+	 * @var	string	forum_id	The forum id from where the topics were moved
+	 * @var	array	forum_ids	Array of the forums where the topics were moved (includes also forum_id)
+	 * @since 3.2.x
```
Comment (dark-1) [inline phpBB/includes/functions_admin.php]: Use :
```php
@since 3.2.9-RC1
```

File: phpBB/includes/functions_admin.php (hunk 1 above)
Comment (OxCom) [reply phpBB/includes/functions_admin.php]: done

File: phpBB/includes/functions_admin.php (hunk 2 above)
Comment (OxCom) [reply phpBB/includes/functions_admin.php]: done

**Comment by dark-1** [comment]
@OxCom ,

You need to remove `[ticket/16143] ` from the commit message:
Here : https://travis-ci.org/phpbb/phpbb/jobs/578252004#L978
from : `[ticket/16143] bump release version for event description`
to : `bump release version for event description`

for Travis-CI to pass...

Best regards 👍

**Comment by mrgoldy** [comment]
No, that's just not true. The `[ticket/16143]` has to be in there..
His older commit messages are wrong. (https://github.com/phpbb/phpbb/pull/5668/commits/d5c77f5548eebd93b0d6a18ec7c929a2cb82a307 and https://github.com/phpbb/phpbb/pull/5668/commits/9eb89a3575516c60d9ae4b57153ce6b57eee78f6 )

The correct syntax is:
> [ticket/12345] A short description
> 
> PHPBB3-12345
> &zwnj;

So the first commit message has to be amended to have a description.
And the second commit message has to be amended to have the `PHPBB3-16143`.

**Comment by dark-1** [comment]
> No, that's just not true. The `[ticket/16143]` has to be in there..
> His older commit messages are wrong. ([d5c77f5](https://github.com/phpbb/phpbb/commit/d5c77f5548eebd93b0d6a18ec7c929a2cb82a307) and [9eb89a3](https://github.com/phpbb/phpbb/commit/9eb89a3575516c60d9ae4b57153ce6b57eee78f6) )
> 
> The correct syntax is:
> 
> > [ticket/12345] A short description
> > PHPBB3-12345
> > ‌
> 
> So the first commit message has to be amended to have a description (a heading).
> And the second commit message has to be amended to have the `PHPBB3-16143`.

Yess!!! you are right.
How come I did not see that....

@OxCom , I am very sorry. 
I this case scenario , I usually close this PR & reopen new one with correct format.

My apologies once again.
Best regards 👍

**Comment by OxCom** [comment]
@dark-1 I will make new one RP.

**Pull request closed without merging**